pkg/admission/server.go

```diff
@@ -96,25 +96,20 @@ var (
 func log500(w http.ResponseWriter, err error) {
 	klog.Errorf("failed to process request: %v\n", err)
 	http.Error(w, err.Error(), http.StatusInternalServerError)
-	return
 }
 
 // ensureKindAdmissionReview check that our admission server is only getting requests
 // for kind AdmissionReview and reject all others
-func ensureKindAdmissionReview(req []byte) bool {
+func ensureKindAdmissionReview(req []byte) (bool, error) {
 	type reqBody struct {
 		Kind  string                 `json:"kind"`
 		Extra map[string]interface{} `json:"-"`
 	}
 	var msg reqBody
-	err := json.Unmarshal(req, &msg)
-	if err != nil {
-		return false
-	}
-	if msg.Kind != "AdmissionReview" {
-		return false
+	if err := json.Unmarshal(req, &msg); err != nil {
+		return false, err
 	}
-	return true
+	return msg.Kind == "AdmissionReview", nil
```

Comment:
I think we should be able to reference a var or constant elsewhere for `"AdmissionReview"` rather than using a string literal for it right? :thinking:

Reply:
Yeah, I thought about this, updated.